lib/cli.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e;
+
+declare -a commands
+declare -a actions
+declare -a hints
+declare -a functions
+
+function register(){
```

Comment:
This register function is really clever and cool, but I'm starting to wonder if we are packing too much functionality into bash scripts. Bash starts out great but has a tendency to get hard to read, update, and test. Thoughts?